cmd/snap/main.go

```diff
@@ -118,7 +118,7 @@ func Parser() *flags.Parser {
 
 		os.Exit(0)
 	}
-	parser := flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash)
+	parser := flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash|flags.PassAfterNonOption)
```

Comment:
By the way, I suspect this was _already_ a bug before. Try for example "snap help foo". It pretends foo isn't there, but it should really error out.